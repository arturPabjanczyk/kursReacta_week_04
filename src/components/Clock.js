import React from "react";
import PropTypes from "prop-types";

function Clock({className, minutes, seconds}) {
    return (
        <h2 className={"clock " + className}>
            <span>Left </span>
            <span className="clock__minutes">{minutes}</span>
            <span className="clock__delimiter">:</span>
            <span className="clock__seconds">{seconds}</span>
        </h2>
    );
}

const NumberOrStringType = PropTypes.oneOfType([PropTypes.number, PropTypes.string]);

Clock.defaultProps = {
    className: "",
    minutes: 20,
    seconds: 48
}

Clock.propTypes = {
    className: PropTypes.string,
    minutes: NonNegativeNumberType.isRequired,
    seconds: NumberOrStringType.isRequired,
}

function NonNegativeNumberType(prop, propName, componentName) {
    if (prop[propName] < 0) {
        return new Error(`Invalid prop '${propName}' issued to component '${componentName}'. It has to be grater or equal to 0.`);
    }
}

export default Clock;