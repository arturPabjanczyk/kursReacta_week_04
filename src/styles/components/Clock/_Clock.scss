.clock {
  color: black;

  &__minutes {
    color: blue;
  }

  &__seconds {
    color: green;
  }

  &__delimiter {
    color: orangered;
  }
}