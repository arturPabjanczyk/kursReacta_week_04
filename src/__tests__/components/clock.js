import React from "react";
import Clock from "../../components/Clock"
import renderer from "react-test-renderer"

var clockRenderer = null;

describe('<Clock />', () => {
    describe('when given minutes and seconds (TestRenderer)', () => {
        beforeEach(() => {
            clockRenderer = renderer.create(
                <Clock minutes={10} seconds={20}/>
            )
        })
        it('shouls render properly', () => {
            expect(clockRenderer.toJSON().type).toMatchSnapshot();
        });
    });
    it('should set className to empty string if not given anything else', () => {
        expect(<Clock minutes={10} seconds={20}/>)
            .toEqual(<Clock className="" minutes={10} seconds={20}/>)
    });
})