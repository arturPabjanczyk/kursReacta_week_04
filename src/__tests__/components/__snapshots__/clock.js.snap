// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Clock /> when given minutes and seconds (TestRenderer) shouls render properly 1`] = `"h2"`;
